tests/unittests/bases/test_ddp.py

```diff
@@ -77,6 +77,40 @@ def _test_ddp_gather_uneven_tensors_multidim(rank: int, worldsize: int = NUM_PRO
         assert (val == torch.ones_like(val)).all()
 
 
+def _test_ddp_gather_autograd_same_shape(rank: int, worldsize: int = NUM_PROCESSES) -> None:
+    tensor = torch.ones(50, requires_grad=True)
```

Comment:
Added docstring and modification `@pytest.mark.skipif(not _TORCH_GREATER_EQUAL_2_1, reason="test only works on newer torch versions")` as per your comment on PT versions. (I just used the 2.1 flag because it was already imported in the same file -- could change it if you want)